Mathlib/CategoryTheory/Limits/Preserves/Yoneda.lean

```diff
@@ -0,0 +1,78 @@
+/-
+Copyright (c) 2024 Markus Himmel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Markus Himmel
+-/
+import Mathlib.CategoryTheory.Limits.Preserves.Ulift
+import Mathlib.CategoryTheory.Limits.FunctorToTypes
+
+/-!
+# Yoneda preserves certain colimits
+
+Given a bifunctor `F : J ⥤ Cᵒᵖ ⥤ Type v`, we prove the isomorphism
+`Hom(YX, colim_j F(j, -)) ≅ colim_j Hom(YX, F(j, -))`, where `Y` is the Yoneda embedding.
+
+We state this in two ways. One is functorial in `X` and stated as a natural isomorphism of functors
+`yoneda.op ⋙ yoneda.obj (colimit F) ≅ yoneda.op ⋙ colimit (F ⋙ yoneda)`, and from this we
+deduce the more traditional preservation statement
+`PreservesColimit F (coyoneda.obj (op (yoneda.obj X)))`.
+
+The proof combines the Yoneda lemma with the fact that colimits in functor categories are computed
+pointwise.
+
+## See also
+
+There is also a relative version of this statement where `F : J ⥤ Over A` for some presheaf
+`A`, see `CategoryTheory.Comma.Presheaf`.
+
+-/
+
+universe v v₁ v₂ v₃ u u₁ u₂ u₃
+
+namespace CategoryTheory
+
+open CategoryTheory.Limits Opposite
+
+variable {C : Type u} [Category.{v} C]
+
+variable {J : Type v} [SmallCategory J] (F : J ⥤ Cᵒᵖ ⥤ Type v)
```

Comment:
Is there a way to relax the universe assumptions on `J` by requesting that certain limits/colimits of shape `J` exist?

Reply:
I made an attempt at this, what do you think?